### PR TITLE
Convert to sdk

### PR DESCRIPTION
**Pull request by ndowmon**
Converted from `@jupiterone/juptier-managed-integration-sdk` to open-source `@jupiterone/integration-sdk-*`.

I looked into any additional projects that still have not been converted:
1. jupiter-integration-aws
2. graph-hackerone
3. graph-openshift
4. graph-threatstack
5. graph-veracode
6. graph-wazuh
7. graph-whitehat
8. graph-zeit

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    resourceName: "Device Sensor Agent",
+    _type: "cbdefense_sensor",
+    _class: ["HostAgent"],
+  },
+  ALERT: {
+    resourceName: "Alert",
+    _type: "cbdefense_alert",
+    _class: ["Finding"],
+  },
+};
```
Comment (aiwilliams) [inline src/constants.ts:31]: Wow, lots of different `_type` value patterns eh? Perhaps we can make these consistent in later releases since with the new SDK and synchronizer it's going to do a DIFF mode sync?

**Review by aiwilliams — DISMISSED** [review]

**Comment by ndowmon** [comment]
Notes from diffing the old and new integrations:

1. The old integration does not seem to produce `carbonblack_psc_account_has_cbdefense_sensor` relationships in J1.

**Comment by ndowmon** [comment]
I diffed the outputs of the old and new integrations using the following two configurations, set up with help from @erichs:
1. https://j1.apps.us.jupiterone.io/integrations/cbdefense/configuration/062df614-dd82-418c-b1de-7340c1195613
2. https://j1.apps.us.jupiterone.io/integrations/cbdefense/configuration/7ef06ac5-f2e6-4d75-b326-9e825b35e9a4

---

I did some manual clean-up first, before diffing. 
1. I noticed that entities and some relationships were tagged with `tag.AccountName: false` by what seems like a bug in the persister. I deleted this property everywhere it showed up, which was essentially everywhere except for in the _new_ mapped relationships.
2. I noticed early on that there didn't seem to be any `carbonblack_psc_account_has_cbdefense_sensor` relationships in the old integration, so I chose to remove them from the new integration. More on that later.
3. The sync API's key generation for mapped relationships behaves differently than that of the managed SDK, so I manually edited mapped relationship keys to match. An example:
    ```diff
    - db631a33-6824-4c5e-a5c1-9aa9e213fb5e:PROTECTS:user_endpoint:88:66:5a:53:7a:87:cbdefense-sensor-58770498|protects|device-joshuas-macbook-pro-2
    + cbdefense-sensor-58770498|protects|device-joshuas-macbook-pro-2
    ````

---

After the clean-up mentioned above, here's the result of `yarn j1-integration diff`:

```diff
➜  graph-cbdefense git:(convert-to-sdk) ✗ yarn j1-integration diff old-no-account-name.json new-mapped-relationships-no-account-name.json 
yarn run v1.22.10
$ /Users/nickdowmon/projects/github/graph-cbdefense/node_modules/.bin/j1-integration diff old-no-account-name.json new-mapped-relationships-no-account-name.json
--- ENTITY DIFF ---
 {
+  cb_endpoint_protection-23830: {
+    _type: [
+      "cb_endpoint_protection"
+    ]
+    _class: [
+      "Service"
+    ]
+    _source: "integration-managed"
+    displayName: "CB Endpoint Protection Service"
+    properties: {
+      name: "CB Endpoint Protection Service"
+      category: [
+        "software"
+        "other"
+      ]
+      function: "monitoring"
+      endpoints: "https://defense-prod05.conferdeploy.net"
+    }
+  }
+  carbonblack-account-23830: {
+    _type: [
+      "carbonblack_psc_account"
+    ]
+    _class: [
+      "Account"
+    ]
+    _source: "integration-managed"
+    displayName: "jupiterone.com"
+    properties: {
+      accountId: 23830
+      name: "jupiterone.com"
+      organization: "jupiterone"
+      webLink: "https://defense-prod05.conferdeploy.net"
+    }
+  }
+  mapper:root: {
+    _type: [
+      "root"
+    ]
+    _class: [
+      "Organization"
+      "Root"
+    ]
+    _source: "system-mapper"
+    displayName: "j1"
+    properties: {
+      name: "JupiterOne, Inc."
+      website: "https://jupiterone.com/"
+      emailDomain: "jupiterone.com"
+    }
+  }
+  Vendor:Carbon Black: {
+    _type: [
+      "carbonblack"
+    ]
+    _class: [
+      "Vendor"
+    ]
+    _source: "system-mapper"
+    displayName: "Carbon Black"
+    properties: {
+      name: "Carbon Black"
+    }
+  }
   cbdefense-sensor-46171184: {
     properties: {
-      lastContactTime: "2021-10-01T15:39:36.925Z"
+      lastContactTime: "2021-10-01T15:23:40.778Z"
-      lastSeenOn: "2021-10-01T15:39:36.925Z"
+      lastSeenOn: "2021-10-01T15:23:40.778Z"
     }
   }
   cbdefense-sensor-51708845: {
     properties: {
-      lastContactTime: "2021-10-01T15:41:17.793Z"
+      lastContactTime: "2021-09-30T16:32:12.623Z"
-      lastExternalIpAddress: "76.174.176.9"
+      lastExternalIpAddress: "174.247.3.51"
-      lastInternalIpAddress: "192.168.1.167"
+      lastInternalIpAddress: "172.20.10.6"
-      lastReportedTime: "2021-10-01T15:41:17.668Z"
+      lastReportedTime: "2021-09-30T16:32:11.135Z"
       sensorStates: [
         "ACTIVE"
+        "SENSOR_SHUTDOWN"
         "LIVE_RESPONSE_NOT_RUNNING"
         "LIVE_RESPONSE_NOT_KILLED"
         "LIVE_RESPONSE_ENABLED"
         "FULL_DISK_ACCESS_NOT_GRANTED"
         "DRIVER_USERSPACE"
       ]
-      lastSeenOn: "2021-10-01T15:41:17.793Z"
+      lastSeenOn: "2021-09-30T16:32:12.623Z"
     }
   }
   cbdefense-sensor-40004077: {
     properties: {
-      lastContactTime: "2021-10-01T15:39:42.717Z"
+      lastContactTime: "2021-10-01T15:23:56.131Z"
-      lastSeenOn: "2021-10-01T15:39:42.717Z"
+      lastSeenOn: "2021-10-01T15:23:56.131Z"
     }
   }
   cbdefense-sensor-36895322: {
     properties: {
-      lastContactTime: "2021-10-01T15:39:27.323Z"
+      lastContactTime: "2021-10-01T15:24:17.575Z"
-      lastSeenOn: "2021-10-01T15:39:27.323Z"
+      lastSeenOn: "2021-10-01T15:24:17.575Z"
     }
   }
   cbdefense-sensor-57645791: {
     properties: {
-      lastContactTime: "2021-10-01T15:40:24.661Z"
+      lastContactTime: "2021-10-01T15:37:24.037Z"
-      lastReportedTime: "2021-10-01T15:39:24.201Z"
+      lastReportedTime: "2021-10-01T15:37:23.953Z"
       sensorStates: [
         "LIVE_RESPONSE_NOT_RUNNING"
         "LIVE_RESPONSE_NOT_KILLED"
         "LIVE_RESPONSE_ENABLED"
-        "FULL_DISK_ACCESS_NOT_GRANTED"
         "DRIVER_USERSPACE"
       ]
-      lastSeenOn: "2021-10-01T15:40:24.661Z"
+      lastSeenOn: "2021-10-01T15:37:24.037Z"
     }
   }
   cbdefense-sensor-46173434: {
     properties: {
-      lastContactTime: "2021-10-01T15:39:44.664Z"
+      lastContactTime: "2021-10-01T15:24:17.836Z"
-      lastSeenOn: "2021-10-01T15:39:44.664Z"
+      lastSeenOn: "2021-10-01T15:24:17.836Z"
     }
   }
   cbdefense-sensor-36867243: {
     properties: {
-      lastContactTime: "2021-10-01T15:39:10.839Z"
+      lastContactTime: "2021-10-01T15:23:10.930Z"
-      lastSeenOn: "2021-10-01T15:39:10.839Z"
+      lastSeenOn: "2021-10-01T15:23:10.930Z"
     }
   }
   cbdefense-sensor-58817790: {
     properties: {
-      lastContactTime: "2021-10-01T15:39:52.121Z"
+      lastContactTime: "2021-10-01T15:37:42.285Z"
-      lastReportedTime: "2021-10-01T15:39:51.635Z"
+      lastReportedTime: "2021-10-01T15:37:42.097Z"
       sensorStates: [
         "LIVE_RESPONSE_NOT_RUNNING"
         "LIVE_RESPONSE_NOT_KILLED"
         "LIVE_RESPONSE_ENABLED"
-        "FULL_DISK_ACCESS_NOT_GRANTED"
         "DRIVER_USERSPACE"
       ]
-      lastSeenOn: "2021-10-01T15:39:52.121Z"
+      lastSeenOn: "2021-10-01T15:37:42.285Z"
     }
   }
   cbdefense-sensor-41528130: {
     properties: {
-      lastContactTime: "2021-10-01T15:40:58.852Z"
+      lastContactTime: "2021-10-01T15:25:38.827Z"
-      lastSeenOn: "2021-10-01T15:40:58.852Z"
+      lastSeenOn: "2021-10-01T15:25:38.827Z"
     }
   }
   cbdefense-sensor-37196276: {
     properties: {
-      lastContactTime: "2021-10-01T15:40:30.063Z"
+      lastContactTime: "2021-10-01T15:24:49.941Z"
-      lastSeenOn: "2021-10-01T15:40:30.063Z"
+      lastSeenOn: "2021-10-01T15:24:49.941Z"
     }
   }
   cbdefense-sensor-36282582: {
     properties: {
-      lastContactTime: "2021-10-01T15:39:38.244Z"
+      lastContactTime: "2021-10-01T15:24:30.950Z"
-      lastSeenOn: "2021-10-01T15:39:38.244Z"
+      lastSeenOn: "2021-10-01T15:24:30.950Z"
     }
   }
   cbdefense-sensor-36281083: {
     properties: {
-      lastContactTime: "2021-10-01T15:40:04.609Z"
+      lastContactTime: "2021-10-01T15:23:56.145Z"
-      lastSeenOn: "2021-10-01T15:40:04.609Z"
+      lastSeenOn: "2021-10-01T15:23:56.145Z"
     }
   }
   cbdefense-sensor-52474447: {
     properties: {
-      lastContactTime: "2021-10-01T15:41:18.991Z"
+      lastContactTime: "2021-10-01T15:24:14.134Z"
-      lastSeenOn: "2021-10-01T15:41:18.991Z"
+      lastSeenOn: "2021-10-01T15:24:14.134Z"
     }
   }
 }

--- RELATIONSHIP DIFF ---
 {
+  carbonblack-account-23830_has_cb_endpoint_protection-23830: {
+    _type: "carbonblack_psc_account_has_cb_endpoint_protection_service"
+    _class: "HAS"
+    _source: "integration-managed"
+    displayName: "HAS"
+    _fromEntityKey: "carbonblack-account-23830"
+    _toEntityKey: "cb_endpoint_protection-23830"
+    properties: {
+    }
+  }
+  b0a68645-e81b-4679-8733-37608eb1df01:OWNS:mapper:root: {
+    _type: "mapped_relationship"
+    _class: "OWNS"
+    _source: "system-mapper"
+    _fromEntityKey: "mapper:root"
+    _toEntityKey: "carbonblack-account-23830"
+    properties: {
+    }
+  }
+  b0a68645-e81b-4679-8733-37608eb1df01:HOSTS:Vendor:Carbon Black: {
+    _type: "mapped_relationship"
+    _class: "HOSTS"
+    _source: "system-mapper"
+    _fromEntityKey: "Vendor:Carbon Black"
+    _toEntityKey: "carbonblack-account-23830"
+    properties: {
+    }
+  }
 }
```

--- 

The entity diff shows that
1. the old integration apparently never did create those account and service entities. That explains all of the missing `carbonblack_psc_account_has_cbdefense_sensor` relationships.
2. the old integration also did not produce global mapping rules to `ROOT` and `carbonblack`. 
3. There are some property differences between the old and new `cbdefense_sensor` entities owing to the fact that the old and new integrations were not invoked at the same moment.

The relationship diff is a result of the old integration not having the carbonblack `account` or `service` entities. 

---

tl;dr the only functional diffs between the old and new integration are that the old integration seems to have had a bug that prevented it from producing the `account` and `service` entities as well as any direct relationships that would have linked to `account` and `service`. The conversion otherwise produces the same result and is good to deploy.

**Review by mknoedel — APPROVED** [review]

File: docs/jupiterone.md [hunk 2]
```diff
@@ -82,29 +85,54 @@ parameters to the integration instance configuration:
...
 
 ### Relationships
 
 The following relationships are created/mapped:
 
-| Relationships                                              |
-| ---------------------------------------------------------- |
-| `carbonblack_psc_account` **HAS** `cbdefense_sensor`       | 
-| `carbonblack_psc_account` **HAS** `cb_endpoint_protection` | 
-| `cbdefense_sensor` **ASSIGNED** `cb_sensor_policy`         |
```
[the remark this answers is not in the document]
Comment (ndowmon) [reply docs/jupiterone.md:106]: IDK why this relationship was defined in this file, but there was never any `cb_sensor_policy` produced from this integration
